pkg/cli/image/info/info.go

```diff
@@ -435,12 +404,67 @@ func (o *ImageRetriever) Run() error {
 						Config:        imageConfig,
 						Layers:        layers,
 						Manifest:      srcManifest,
-					}, manifestErr); err != nil {
+					}
+					if err = callbackFn(name, imageRetrieved, manifestErr); err != nil {
 						return err
 					}
 				}
 				return nil
 			})
 		}
 	})
+	if err != nil {
+		return nil, err
+	}
+	return imageRetrieved, nil
+}
+
+// ImageRetrieverOrList returns ImageRetriever and with the ManifestListCallbackFn provides a message
+// to distinguish errors between multiple manifests found and no manifests found, and lists os/arch options for manifestList upon error
+func (o *InfoOptions) ImageRetrieverOrList(src, filterByOS string) (*ImageRetriever, error) {
```

Comment:
If you extract the lower method this larger function should go back to being inlined (you're just passing a standard function to ManifestListCallback at that point)

Reply:
I'm not completely opposed to this method being extracted, it's just going to have a lot of arguments and that's a bit of a code smell.

Reply:
no reason to extract this, un-extracted it